src/ApplicationLogs/LogReader.cs

```diff
@@ -186,6 +197,33 @@ private void OnCommitting(NeoSystem system, Block block, DataCache snapshot, IRe
             }
         }
 
+        private static void ApplicationEngine_Log(object sender, LogEventArgs args)
+        {
+            if (!Settings.Default.Debug) return;
+
+            var tx = ((Transaction)args.ScriptContainer).Hash;
+
+            byte[] value = _db.TryGet(LogPrefix.Concat(tx.ToArray()).ToArray());
+
+            JArray logList = null;
+            if (value is null)
+            {
+                logList = new JArray();
+            }
+            else
+            {
+                logList = (JArray)JToken.Parse(Neo.Utility.StrictUTF8.GetString(value));
+            }
+            var logJson = new JObject
+            {
+                ["contract"] = args.ScriptHash.ToString(),
+                ["message"] = args.Message
+            };
+            logList?.Add(logJson);
```

Comment:
Why do you need `?` here? `logList` is always non-null.

Reply:
> Why do you need `?` here? `logList` is always non-null.

It is possible since in the `else` block, the assignment might be null.